src/hotspot/share/utilities/globalDefinitions.hpp

```diff
@@ -188,6 +188,7 @@ FORBID_C_FUNCTION(void exit(int), "use os::exit");
 FORBID_C_FUNCTION(void _exit(int), "use os::exit");
 FORBID_C_FUNCTION(char* strerror(int), "use os::strerror");
 FORBID_C_FUNCTION(char* strtok(char*, const char*), "use strtok_r");
+FORBID_C_FUNCTION(int sprintf(char*, const char*, ...), "use os::snprintf");
```

Comment:
I have to wonder whether this actually works too. Perhaps @kbarrett can comment?

Reply:
For the record I tested this standalone and it works fine in a C++ program.